e2e-polybft/consensus_test.go

```diff
@@ -189,3 +189,91 @@ func TestE2E_Consensus_RegisterValidator(t *testing.T) {
 	t.Logf("New validator rewards=%s\n", rewards)
 	require.True(t, rewards.Cmp(big.NewInt(0)) > 0)
 }
+
+func TestE2E_Consensus_Delegation_Undelegation(t *testing.T) {
+	const (
+		validatorSecrets = "test-chain-1"
+		delegatorSecrets = "test-chain-6"
+		premineBalance   = "0x1B1AE4D6E2EF500000" // 500 native tokens (so that we have enough funds to fund delegator)
+	)
+
+	fundAmountRaw := "0xD8D726B7177A80000" // 250 native tokens
+	fundAmount, err := types.ParseUint256orHex(&fundAmountRaw)
+	require.NoError(t, err)
+
+	cluster := framework.NewTestCluster(t, 5,
+		framework.WithEpochReward(100000),
+		framework.WithPremineValidators(premineBalance))
+
+	// init delegator account
+	require.NoError(t, cluster.InitSecrets(delegatorSecrets, 1))
+	srv := cluster.Servers[0]
+
+	txRelayer, err := txrelayer.NewTxRelayer(txrelayer.WithIPAddress(srv.JSONRPCAddr()))
+	require.NoError(t, err)
+
+	cluster.WaitForBlock(1, 10*time.Second)
+
+	delegatorAcc, err := sidechain.GetAccountFromDir(path.Join(cluster.Config.TmpDir, delegatorSecrets))
+	require.NoError(t, err)
+
+	delegatorAddr := delegatorAcc.Ecdsa.Address()
+
+	validatorAcc, err := sidechain.GetAccountFromDir(path.Join(cluster.Config.TmpDir, validatorSecrets))
+	require.NoError(t, err)
+
+	validatorAddr := validatorAcc.Ecdsa.Address()
+
+	// fund delegator
+	receipt, err := txRelayer.SendTransaction(&ethgo.Transaction{
+		From:  validatorAddr,
+		To:    &delegatorAddr,
+		Value: fundAmount,
+	}, validatorAcc.Ecdsa)
+	require.NoError(t, err)
+	require.Equal(t, uint64(types.ReceiptSuccess), receipt.Status)
+
+	queryDelegator := func(blockNum uint64) (balance *big.Int, reward *big.Int) {
+		var err error
+		balance, err = srv.JSONRPC().Eth().GetBalance(delegatorAddr, ethgo.Latest)
+		require.NoError(t, err)
+		t.Logf("Delegator balance (block %d)=%s\n", blockNum, balance)
+
+		reward, err = sidechain.GetDelegatorReward(validatorAddr, delegatorAddr, txRelayer)
+		require.NoError(t, err)
+		t.Logf("Delegator reward (block %d)=%s\n", blockNum, reward)
+
+		return
+	}
+
+	delegatorBalance, _ := queryDelegator(5)
```

Comment:
Maybe add `waitForBlock(5)` before you query this? Since you are calling this `queryDelegator` with block 5.

Reply:
Good point. In fact, I've added querying for current block number, because previously provided information weren't correct :slightly_smiling_face: 
https://github.com/0xPolygon/polygon-edge/pull/1086/commits/7d638f94a7f9202fc648ca9f6439521d02d41ffc